dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/intermediate/components/IntegerComponent.kt

```diff
@@ -0,0 +1,68 @@
+package org.dataland.frameworktoolbox.intermediate.components
+
+import org.apache.commons.text.StringEscapeUtils
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.UploadCategoryBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+
+/**
+ * A IntegerComponent represents a numeric value from the integer domain
+ */
+open class IntegerComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+    open var constantUnitSuffix: String? = null
+
+    override fun generateDefaultDataModel(dataClassBuilder: DataClassBuilder) {
+        dataClassBuilder.addProperty(
+            this.identifier,
+            documentSupport.getJvmTypeReference(
+                TypeReference("java.math.BigInteger", isNullable),
```

Comment:
For `DecimalComponent` the value `isNullable` is not passed into `TypeReference(...)` but the constant `true`. Why is this different here?

Reply:
I have no idea.  I think we should also write `isNullable` in the DecimalComponent.
Otherwise you cannot make decimal fields non-nullable via the excel column "Manadtory field".

What do you think?

Reply:
I would like to talk about that. In this version you can not control the datapoint to be nullable and the template type to be nullable independently. Maybe I just don't understand something correclty.